error TS4033: Property 'inputRef' of exported interface has or is using private name '__VLS_6'

### Vue - Official extension or vue-tsc version

2.2.2

### VSCode version

-

### Vue version

3.5.13

### TypeScript version

5.7.3

### System Info

```shell

```

### package.json dependencies

```json

```

### Steps to reproduce

```sh
git clone git@github.com:deathmaz/vue-tsc-repro.git
cd vue-tsc-repro
yarn
yarn type-check
```


### What is expected?

No error after running `yarn type-check`

### What is actually happening?

After running `yarn type-check` the following error is thrown:
```sh
src/components/TestComponent.vue:41:18 - error TS4033: Property 'inputRef' of exported interface has or is using private name '__VLS_6'.

41 inputRef: typeof __VLS_6,
                    ~~~~~~~


Found 1 error.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Link to minimal reproduction

https://github.com/deathmaz/vue-tsc-repro

### Any additional comments?

The error is not happening in this case:
```diff
diff --git a/src/components/TestComponent.vue b/src/components/TestComponent.vue
index 77b42af..dea08dd 100644
--- a/src/components/TestComponent.vue
+++ b/src/components/TestComponent.vue
@@ -4,7 +4,6 @@
   <template v-else>
     <component
       :is="'textarea'"
-      ref="inputRef"
     />
   </template>
 </template>
```

and in this case:
```diff
diff --git a/src/components/TestComponent.vue b/src/components/TestComponent.vue
index 77b42af..dde4b52 100644
--- a/src/components/TestComponent.vue
+++ b/src/components/TestComponent.vue
@@ -1,7 +1,5 @@
 <template>
-  <template v-if="false">
-  </template>
-  <template v-else>
+  <template>
     <component
       :is="'textarea'"
       ref="inputRef"
```

--- a/packages/language-core/lib/codegen/template/context.ts
+++ b/packages/language-core/lib/codegen/template/context.ts
@@ -43,6 +43,7 @@ export function createTemplateCodegenContext(options: Pick<TemplateCodegenOption
 		return features;
 	}
 
+	const hoistVars = new Map<string, string>();
 	const localVars = new Map<string, number>();
 	const specialVars = new Set<string>();
 	const accessExternalVariables = new Map<string, Set<number>>();
@@ -67,7 +68,10 @@ export function createTemplateCodegenContext(options: Pick<TemplateCodegenOption
 	const inlayHints: InlayHintInfo[] = [];
 	const bindingAttrLocs: CompilerDOM.SourceLocation[] = [];
 	const inheritedAttrVars = new Set<string>();
-	const templateRefs = new Map<string, [varName: string, offset: number]>();
+	const templateRefs = new Map<string, {
+		varName: string;
+		offset: number;
+	}>();
 
 	return {
 		codeFeatures: new Proxy(codeFeatures, {
@@ -115,6 +119,19 @@ export function createTemplateCodegenContext(options: Pick<TemplateCodegenOption
 		},
 		getInternalVariable: () => {
 			return `__VLS_${variableId++}`;
+		},
+		getHoistVariable: (originalVar: string) => {
+			let name = hoistVars.get(originalVar);
+			if (name === undefined) {
+				hoistVars.set(originalVar, name = `__VLS_${variableId++}`);
+			}
+			return name;
+		},
+		generateHoistVariables: function* () {
+			// trick to avoid TS 4081 (#5186)
+			for (const [originalVar, hoistVar] of hoistVars) {
+				yield `var ${hoistVar} = ${originalVar}${endOfLine}`;
+			}
 		},
 		ignoreError: function* (): Generator<Code> {
 			if (!ignoredError) {

--- a/packages/language-core/lib/codegen/template/element.ts
+++ b/packages/language-core/lib/codegen/template/element.ts
@@ -272,7 +272,10 @@ export function* generateComponent(
 		yield `${endOfLine}`;
 
 		if (refName) {
-			ctx.templateRefs.set(refName, [varName, offset!]);
+			ctx.templateRefs.set(refName, {
+				varName: ctx.getHoistVariable(varName),
+				offset: offset!
+			});
 		}
 		if (isRootNode) {
 			ctx.singleRootElType = `NonNullable<typeof ${varName}>['$el']`;
@@ -358,11 +361,14 @@ export function* generateElement(
 
 	const [refName, offset] = yield* generateVScope(options, ctx, node, node.props);
 	if (refName) {
-		let refValue = `__VLS_nativeElements['${node.tag}']`;
+		let element = `__VLS_nativeElements['${node.tag}']`;
 		if (isVForChild) {
-			refValue = `[${refValue}]`;
+			element = `[${element}]`;
 		}
-		ctx.templateRefs.set(refName, [refValue, offset!]);
+		ctx.templateRefs.set(refName, {
+			varName: element,
+			offset: offset!
+		});
 	}
 	if (ctx.singleRootNode === node) {
 		ctx.singleRootElType = `typeof __VLS_nativeElements['${node.tag}']`;

--- a/packages/language-core/lib/codegen/template/index.ts
+++ b/packages/language-core/lib/codegen/template/index.ts
@@ -45,6 +45,9 @@ export function* generateTemplate(options: TemplateCodegenOptions): Generator<Co
 	}
 
 	yield* generateStyleScopedClassReferences(ctx);
+	yield* ctx.generateAutoImportCompletion();
+	yield* ctx.generateHoistVariables();
+
 	const speicalTypes = [
 		[slotsPropertyName, yield* generateSlots(options, ctx)],
 		['$attrs', yield* generateInheritedAttrs(options, ctx)],
@@ -58,7 +61,6 @@ export function* generateTemplate(options: TemplateCodegenOptions): Generator<Co
 	}
 	yield `} & { [K in keyof import('${options.vueCompilerOptions.lib}').ComponentPublicInstance]: unknown }${endOfLine}`;
 
-	yield* ctx.generateAutoImportCompletion();
 	return ctx;
 }
 
@@ -67,20 +69,6 @@ function* generateSlots(
 	ctx: TemplateCodegenContext
 ): Generator<Code> {
 	if (!options.hasDefineSlots) {
-		const hoistVars = new Map<string, string>();
-
-		// trick to avoid TS 4081 (#5186)
-		for (const slot of ctx.dynamicSlots) {
-			hoistVars.set(slot.expVar, slot.expVar = ctx.getInternalVariable());
-			hoistVars.set(slot.propsVar, slot.propsVar = ctx.getInternalVariable());
-		}
-		for (const slot of ctx.slots) {
-			hoistVars.set(slot.propsVar, slot.propsVar = ctx.getInternalVariable());
-		}
-		for (const [originalVar, hoistVar] of hoistVars) {
-			yield `var ${hoistVar} = ${originalVar}${endOfLine}`;
-		}
-
 		const name = getSlotsPropertyName(options.vueCompilerOptions.target);
 		yield `type __VLS_Slots = __VLS_PrettifyGlobal<__VLS_OmitStringIndex<typeof __VLS_ctx.${name}>`;
 		for (const { expVar, propsVar } of ctx.dynamicSlots) {
@@ -145,7 +133,7 @@ function* generateTemplateRefs(
 	ctx: TemplateCodegenContext
 ): Generator<Code> {
 	yield `type __VLS_TemplateRefs = {${newLine}`;
-	for (const [name, [varName, offset]] of ctx.templateRefs) {
+	for (const [name, { varName, offset }] of ctx.templateRefs) {
 		yield* generateObjectProperty(
 			options,
 			ctx,

--- a/packages/language-core/lib/codegen/template/slotOutlet.ts
+++ b/packages/language-core/lib/codegen/template/slotOutlet.ts
@@ -131,7 +131,7 @@ export function* generateSlotOutlet(
 				offset: nameProp.loc.start.offset + nameProp.loc.source.indexOf(nameProp.value.content, nameProp.name.length),
 				tagRange: [startTagOffset, startTagOffset + node.tag.length],
 				nodeLoc: node.loc,
-				propsVar,
+				propsVar: ctx.getHoistVariable(propsVar),
 			});
 		}
 		else if (
@@ -155,8 +155,8 @@ export function* generateSlotOutlet(
 			);
 			yield `)${endOfLine}`;
 			ctx.dynamicSlots.push({
-				expVar,
-				propsVar,
+				expVar: ctx.getHoistVariable(expVar),
+				propsVar: ctx.getHoistVariable(propsVar),
 			});
 		}
 		else {
@@ -164,7 +164,7 @@ export function* generateSlotOutlet(
 				name: 'default',
 				tagRange: [startTagOffset, startTagEndOffset],
 				nodeLoc: node.loc,
-				propsVar,
+				propsVar: ctx.getHoistVariable(propsVar),
 			});
 		}
 	}

